### PR TITLE
Rewrite C backend now with expressions AND System V message IPC!

### PR DESCRIPTION
**Pull request by professionalmoment**
This doesn't have any way to communicate with the python code implemented yet but that should be done soon.
Make sure you use the new pa-ctl command in the merge_devices.sh script otherwise you will experience the duplicate audio @breaker-of-stone was talking about

### CODE REVIEW

**Comment by professionalmoment** [comment]
note: this should also be portable so if you want to test it out on your machine (without the 4 speaker setup) you totally can including the visualization but I have only tested it on Linux

**Comment by professionalmoment** [comment]
This is means everything (except for effects) is done!

File: examples/audio-play/c backend/config.h [hunk 1]
```diff
@@ -5,6 +5,29 @@
 // 25 ms
 #define LOOP_TARGET_TIME 25000
 
+// MAKE SURE THESE MATCH OR IT WILL CRASH
+#define ZMQ_BIND_ADDR "tcp://*:5555"
+#define ZMQ_BIND_PORT 5555
+
+
+// this list is not final and WILL change
+#define EFFECT_TYPE_STRING_ARRAY {"noEffect", "delay", "distortion", "chorus", "saturation", "phaser", "reverb", "remolo"}
```
Comment (ethancrawford) [inline examples/audio-play/c backend/config.h:14]: typo 😄 
```suggestion
#define EFFECT_TYPE_STRING_ARRAY {"noEffect", "delay", "distortion", "chorus", "saturation", "phaser", "reverb", "tremolo"}
```

Comment (professionalmoment) [reply examples/audio-play/c backend/config.h:14]: whoops. I do a lot of those

Comment (ethancrawford) [reply examples/audio-play/c backend/config.h]: Eh, we all do it 😆

**Pull request merged**